### PR TITLE
Add messages sent metrics to kathy

### LINKED ISSUE #739: Push metrics from kathy so we can more easily determine if there are failures in sending messages

### LINKED ISSUE #744: Enable Ethereum in Kathy

### PR DESCRIPTION
**Pull request by mattiecnvr**
Main interest here is actually the failures, but this adds metrics so we can track the messages sent and messages we failed to send by origin and remote. This will make it easier to alert if there is an issue since all we will need to do is check if there are reported errors.

This will also re-enable ethereuem for mainnet now that we will get value in tracking it.

Also kinda just stuck in a change to the gitignore because I got tired of these files sitting as unincluded in the git repo.

CC @yorhodes 

Closes #739 
Closes #744 
Touches #745

### CODE REVIEW

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 1]
```diff
@@ -57,6 +89,32 @@ async function sendMessage(
...
+
+async function submitMetrics() {
+  const gateway = getPushGateway();
+  if (!gateway) return;
+
+  const { resp, body } = await gateway.push({ jobName: 'relayer_funder' });
+  const statusCode =
+    typeof resp == 'object' && resp != null && 'statusCode' in resp
+      ? (resp as any).statusCode
+      : 'unknown';
```
Comment (yorhodes) [inline typescript/infra/scripts/helloworld/kathy.ts:112]:
```suggestion
  const statusCode = resp?.statusCode ?? 'unknown';
```

**Review by yorhodes — COMMENTED** [review]
few comments

File: solidity/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+*/artifacts
```
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply solidity/.gitignore:1]: It wasn't being ignored on mine.

File: typescript/infra/config/environments/testnet2/helloworld.ts [hunk 3]
```diff
@@ -14,5 +14,7 @@ export const helloWorld: HelloWorldConfig<TestnetChains> = {
     chainsToSkip: [],
     runEnv: environment,
     namespace: environment,
+    prometheusPushGateway:
+      'http://prometheus-pushgateway.monitoring.svc.cluster.local:9091',
```
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply typescript/infra/config/environments/testnet2/helloworld.ts:18]: It's not the same cluster. Both mainnet and testnet will point to the pushgateway that is in their cluster, they are just mirrored so it's not super obvious because the domain names end up being the same.

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 4]
```diff
@@ -6,8 +8,29 @@ import { getCoreEnvironmentConfig, getEnvironment } from '../utils';
...
+  help: 'Test messages which have been sent with',
+  registers: [metricsRegister],
+  labelNames: [
+    'origin',
+    'remote',
+    'status',
+    ...(Object.keys(constMetricLabels) as (keyof typeof constMetricLabels)[]),
+  ],
+});
+metricsRegister.registerMetric(messagesCount);
```
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply typescript/infra/scripts/helloworld/kathy.ts:29]: 1) Metrics really should be static because they are a global count and passing them around is annoying
2) `registerMetric` does not need to run after `abacus_deployment`, all this does in the static context is pull the label lame (hence `Object.keys` and not `Object.values`).

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 5]
```diff
@@ -6,8 +8,29 @@ import { getCoreEnvironmentConfig, getEnvironment } from '../utils';
...
+const metricsRegister = new Registry();
+const messagesCount = new Counter({
+  name: 'abacus_kathy_messages',
+  help: 'Test messages which have been sent with',
+  registers: [metricsRegister],
+  labelNames: [
+    'origin',
+    'remote',
+    'status',
+    ...(Object.keys(constMetricLabels) as (keyof typeof constMetricLabels)[]),
```
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply typescript/infra/scripts/helloworld/kathy.ts:26]: Yes, we need the cast. This is for the strongly typed labels to work.

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 6]
```diff
@@ -26,21 +49,30 @@ async function main() {
...
         );
         failureOccurred = true;
+        messagesCount.labels({ ...labels, status: 'failure' }).inc();
       }
       // Sleep 500ms to avoid race conditions where nonces are reused
       await sleep(500);
     }
   }
 
+  await submitMetrics();
```
[the remark this answers is not in the document]
Comment (mattiecnvr) [reply typescript/infra/scripts/helloworld/kathy.ts:74]: We could submit before sleeping, but then we are doing a lot more submissions since that is a loop and it is fairly traditional with push metrics to only report on job completion so this seems fine.

File: .gitignore [hunk 7]
```diff
@@ -7,6 +7,7 @@ typescript/abacus-deploy/.env
 **/tmp.*
 **/*.swp
 **/*.swo
+**/dist
```
[the remark this answers is not in the document]
Comment (yorhodes) [reply .gitignore:10]: `typescript/deploy` was removed in https://github.com/abacus-network/abacus-monorepo/pull/643

File: typescript/infra/scripts/helloworld/kathy.ts (hunk 1 above)
Comment (yorhodes) [reply typescript/infra/scripts/helloworld/kathy.ts:112]: the ternary is very hard to read
are there no types for push https://github.com/prometheus/pushgateway?

File: solidity/.gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (nambrot) [reply solidity/.gitignore:1]: +1 to keeping a root .gitignore

File: typescript/infra/scripts/helloworld/kathy.ts [hunk 8]
```diff
@@ -57,6 +89,32 @@ async function sendMessage(
   console.log(JSON.stringify(receipt.events || receipt.logs));
 }
 
+function getPushGateway(): Pushgateway | null {
```
Comment (tkporter) [inline typescript/infra/scripts/helloworld/kathy.ts:92]: Can we de-dup some of this and `submitMetrics` with the code in the relayer funder, i.e. make them available as common utils or something?
While we're at it, it'd be nice to also use the structured logging here that we do in the relayer funder. It'd make it a lot nicer to browse these logs when they're deployed

File: .gitignore [hunk 9]
```diff
@@ -14,10 +14,8 @@ rust/tmp.env
 tmp.env
 .DS_STORE
 
-typescript/*/.env
 typescript/*/node_modules
-typescript/**/tsconfig.tsbuildinfo
-**/**/tsconfig.tsbuildinfo
```
Comment (tkporter) [inline .gitignore:20]: why the changes here? Don't really remember the exact situation where `tsconfig.tsbuildinfo` files are created to be fair but I know they shouldn't be committed

**Review by tkporter — APPROVED** [review]
would ideally like to see some deduplication given a lot of this is pretty similar to the relayer funder

File: .gitignore (hunk 9 above)
Comment (mattiecnvr) [reply .gitignore:20]: I just deduplicated a bit, read the full gitignore and you'll see they were already getting covered
